src/common/imagwebp.cpp

```diff
@@ -0,0 +1,346 @@
+            }
+            break;
+        }
+
+        wxWebPAnimationFrame frame;
+        frame.image.Create(anim_info.canvas_width, anim_info.canvas_height, false);
+        frame.bgColour.SetRGBA(anim_info.bgcolor);
+        rgbaToImage(&frame.image, buf);
+        frame.duration = timestamp - prevTimestamp;
+        prevTimestamp = timestamp;
+        frames.push_back(std::move(frame));
+    }
+
+    WebPAnimDecoderReset(decoder.get());
+
+    return true;
+}
+
+bool wxWEBPHandler::SaveFile(wxImage *image, wxOutputStream& stream, bool)
+{
+    float quality_factor = 90; // if you change this, update the documentation, too
+    if (image->HasOption(wxIMAGE_OPTION_WEBP_QUALITY))
+        quality_factor = image->GetOptionInt(wxIMAGE_OPTION_WEBP_QUALITY);
+    bool lossless = image->GetOptionInt(wxIMAGE_OPTION_WEBP_LOSSLESS) == 2;
+
+    size_t output_size = 0;
+    uint8_t * output = nullptr;
+    unsigned char * rgb = image->GetData();
+    if (image->HasAlpha())
+    {
+        unsigned char* alpha = image->GetAlpha();
+        size_t pixel_count = (size_t)image->GetWidth() * (size_t)image->GetHeight();
+        std::vector<unsigned char> rgba(pixel_count * 4);
+        size_t rgba_index = 0, rgb_index = 0, alpha_index = 0;
+        for (size_t pixel_counter = 0; pixel_counter < pixel_count; pixel_counter++)
+        {
+            rgba[rgba_index++] = rgb[rgb_index++]; // R
+            rgba[rgba_index++] = rgb[rgb_index++]; // G
+            rgba[rgba_index++] = rgb[rgb_index++]; // B
+            rgba[rgba_index++] = alpha[alpha_index++]; // A
+        }
+        int stride = image->GetWidth() * 4; // stride is the "width" of a "line" in bytes
+        if (lossless)
+            output_size = WebPEncodeLosslessRGBA(rgba.data(), image->GetWidth(), image->GetHeight(), stride, &output);
+        else
+            output_size = WebPEncodeRGBA(rgba.data(), image->GetWidth(), image->GetHeight(), stride, quality_factor, &output);
+    }
+    else
+    {
+        int stride = image->GetWidth() * 3; // stride is the "width" of a "line" in bytes
+        if (lossless)
+            output_size = WebPEncodeLosslessRGB(rgb, image->GetWidth(), image->GetHeight(), stride, &output);
+        else
+            output_size = WebPEncodeRGB(rgb, image->GetWidth(), image->GetHeight(), stride, quality_factor, &output);
+    }
+
+    stream.WriteAll(output, output_size);
+    WebPFree(output);
+    return (output_size > 0 && stream.LastWrite() == output_size);
+}
+
+int wxWEBPHandler::DoGetImageCount(wxInputStream & stream)
+{
+    int frame_count = -1;
+    WebPDemuxerPtr demux = CreateDemuxer(stream);
+    if (demux != nullptr)
+    {
+        frame_count = WebPDemuxGetI(demux.get(), WEBP_FF_FRAME_COUNT);
+    }
+    return frame_count;
+}
+
+bool wxWEBPHandler::DoCanRead(wxInputStream& stream)
+{
+    // check header according to https://developers.google.com/speed/webp/docs/riff_container
+    const std::string riff = "RIFF";
```

Comment:
Would it not be better to use `constexpr char*` instead of using and creating four `std::string` here?